### PR TITLE
Feature/gtm

### PR DESCRIPTION
**Pull request by andreldsa**
#### What is the purpose of this pull request?
MVP of Google Tag Manager app

#### What problem is this solving?
Extracting the GTM code from the Store and transferring to a modular app.

#### How should this be manually tested?
https://pixeeel--storecomponents.myvtex.com/

#### Types of changes
* New feature (a non-breaking change which adds functionality)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,34 @@
-# google-tagmanager
-VTEX app that integrate with Google Tag Manager
+# How to add to your Store
+
+1 - Open the VTEX APP Store and install the app on your store.
+
+2 - Add the following line to your store manifest:
```
Comment (brunojdo) [inline README.md]: How to configure the GTM ID?

File: manifest.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "vtex.store": "1.x"
+  },
+  "settingsSchema": {
+    "title": "Google Tag Manager",
+    "type": "object",
+    "properties": {
+      "gtmId": {
+        "title": "Google Tag Manager",
+        "type": "string",
+        "description": "Enter the ID (AW-XXXX) from your Google Tag Manager"
```
Comment (brunojdo) [inline manifest.json]: `Enter the ID (GTM-XXXX) ...`

File: react/index.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      event: 'productView',
+      send_to: this.gtmId,
+      sku_id: skuId
+    })
+  }
+
+  componentDidMount() {
+    window.dataLayer = window.dataLayer || []
+    this.gtm({js: new Date()})
+    this.gtm({config: this.gtmID})
```
Comment (gugabribeiro) [inline react/index.js]: Change `gtmID` to `gtmId`.

File: react/index.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  }
+  
+  gtm(data) {
+    window.dataLayer.push(data)
+  }
+  
+  get gtmId() {
+    const { gtmId } = this.context.getSettings(APP_LOCATOR) || {} 
+    if (!gtmId) {
+      console.warn("Warning: The GTM ID isn't set on your environment.")
```
Comment (brunojdo) [inline react/index.js]: Do like I did in the vtex.store

```
const GTM_UNDEFINED = `No Google Tag Manager ID is defined. Take a look at:\
https://${global.__RUNTIME__.workspace}--${global.__RUNTIME__.account}.myvtex.com/admin/apps/vtex.google-tag-manager/setup`
```
and `console.warn(GTM_UNDEFINED)`

File: react/index.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
+import { Component } from 'react'
+import { withRuntimeContext, Helmet } from 'render'
+import { Pixel } from 'vtex.store/PixelContext'
+import { gtmScript, gtmFrame } from './scripts/gtm'
+
+const APP_LOCATOR = 'vtex.google-tag-manager'
+
+const GTM_UNDEFINED = ({ workspace, account }) => (
+  `No Google Tag Manager ID is defined. Take a look at:\
+  https://${workspace}--${account}.myvtex.com/admin/apps/vtex.store/setup`
```
Comment (brunojdo) [inline react/index.js]: `  https://${workspace}--${account}.myvtex.com/admin/apps/${APP_LOCATOR}/setup
`

**Pull request merged**